tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -147,6 +147,76 @@ public function test_register() {
 		);
 	}
 
+	/**
+	 * @dataProvider analytics_sharing_settings_data_provider
+	 * @param array $sharing_settings
+	 * @param array $expected
+	 */
+	public function test_register__replicate_analytics_sharing_settings( $sharing_settings, $expected ) {
+		remove_all_filters( 'option_' . Module_Sharing_Settings::OPTION );
+		$this->assertFalse( has_filter( 'option_' . Module_Sharing_Settings::OPTION ) );
+
+		$this->analytics->register();
+
+		$this->assertTrue( has_filter( 'option_' . Module_Sharing_Settings::OPTION ) );
+
+		update_option( Module_Sharing_Settings::OPTION, $sharing_settings );
+		$this->assertEquals( $expected, get_option( Module_Sharing_Settings::OPTION ) );
+	}
+
+	public function analytics_sharing_settings_data_provider() {
+		$initial_sharing_settings                                     = array(
+			'search-console' => array(
+				'sharedRoles' => array( 'contributor', 'administrator' ),
+				'management'  => 'all_admins',
+			),
+		);
+		$sharing_settings_with_analytics                              = array_merge(
+			$initial_sharing_settings,
+			array(
+				'analytics' => array(
+					'sharedRoles' => array( 'editor', 'administrator' ),
+					'management'  => 'owner',
+				),
+			)
+		);
+		$sharing_settings_with_both_analytics                         = array_merge(
+			$sharing_settings_with_analytics,
+			array(
+				'analytics-4' => array(
+					'sharedRoles' => array( 'editor', 'administrator' ),
+					'management'  => 'owner',
+				),
+			)
+		);
+		$sharing_settings_with_both_analytics_with_different_settings = array_merge(
+			$sharing_settings_with_analytics,
+			array(
+				'analytics-4' => array(
+					'sharedRoles' => array( 'contributor' ),
+					'management'  => 'all_admins',
+				),
+			)
+		);
+		$modified_settings_with_both_analytics                        = $sharing_settings_with_both_analytics;
+		$modified_settings_with_both_analytics['analytics']['sharedRoles'] = array( 'contributor' );
+
+		return array(
+			'Analytics and Analytics-4 both not set' => array(
+				$initial_sharing_settings,
+				$initial_sharing_settings,
+			),
+			'Analytics set and Analytics-4 not set'  => array(
+				$sharing_settings_with_analytics,
+				$sharing_settings_with_both_analytics,
+			),
+			'Analytics and Analytics-4 both set'     => array(
+				$sharing_settings_with_both_analytics_with_different_settings,
```

Comment:
Hi @jimmymadon, looking at the previous implementation of this test, it was evident that the test could in fact provide a false impression of the behaviour - with both Analytics and Analytics 4 having the same sharing settings, it wasn't verifying that the Analytics settings were _not_ replicated in this scenario, as the settings were the same anyway. In order to save time, I've pushed an update which provides different settings for the two modules, so it can be seen that the Analytics settings were definitely not replicated.

Please review the change that I've made and if you're happy with it, this is good to merge as far as I'm concerned.

Reply:
So sorry to miss that @jimmymadon!

Yes, I like your suggestion to make things more verbose, should make it a bit clearer going forward so people like me don't miss it :facepalm: